go/pkg/client/capabilities.go

```diff
@@ -28,6 +28,18 @@ func (c *Client) CheckCapabilities(ctx context.Context) (err error) {
 	if c.serverCaps.CacheCapabilities != nil {
 		c.MaxBatchSize = MaxBatchSize(c.serverCaps.CacheCapabilities.MaxBatchTotalSizeBytes)
 	}
+	if c.serverCaps.CacheCapabilities.SupportedCompressors != nil && c.CompressedBytestreamThreshold >= 0  {
+		foundZstd := false
+		for _, sComp := range c.serverCaps.CacheCapabilities.SupportedCompressors {
+			if sComp == repb.Compressor_ZSTD {
+				foundZstd = true
+				break
+			}
+		}
+		if !foundZstd {
+			return errors.New("zstd compression is not supported by server")
```

Comment:
should it be returned also when `CompressedBytestreamThreshold >= 0` and `SupportedCompressors = nil`?

Reply:
Yup -fixed.